code/modules/mob/living/carbon/xenomorph/Facehuggers.dm

```diff
@@ -258,18 +251,28 @@
 			addtimer(CALLBACK(src, .proc/fast_activate), 1.5 SECONDS)
 
 	else
+		for(var/mob/M in loc)
+			if(CanHug(M))
+				addtimer(CALLBACK(src, .proc/fast_facehug, M), 1.5 SECONDS)
+				return
 		addtimer(CALLBACK(src, .proc/fast_activate), rand(MIN_ACTIVE_TIME,MAX_ACTIVE_TIME))
 		return ..()
 
+/obj/item/clothing/mask/facehugger/proc/fast_facehug(mob/M)
+	if(!QDELETED(M) && Adjacent(M) && CanHug(M) && isturf(M.loc))
```

Comment:
This is called with a `for(var/mob/M in loc)`, which means the same facehugger will try to attach itself to as many viable mobs there are in the tile. What's preventing it from hugging more than one?

Reply:
it returns when it finds a potential victim, so to avoid looping furthermore.